### DISCUSSION TITLE
For SSA of an e-commerce application, what is the need of Nextjs?

### DISCUSSION THREAD

**Discussion by biplobmahadi** [vercel/next.js#10708] [Help]
I want to make a Single Site Application which is an e-commerce application. For this I want use **Django, Django Rest** for API endpoints as backend. And for frontend I want to use **React, Redux, Axios, React-Router, Antd**. 
So for professionally deploy this app, what is need of **Nextjs or Gatsby**? I want to scale my application for million users.
Please help & Thanks.

**Comment by leerob** [comment]
Let me try to rephrase and summarize your requirements:

- Static generated e-commerce site
- Backend/API routes needed
- Frontend using React

Should you use Next.js or Gatsby? In my opinion, Next.js seems like a better fit. Here's why:

- **Hybrid app support.** Whether you need to use SSR (server-side rendering) or SSG (static site generation), Next.js can support it out of the box. Both approaches have a time and place inside e-commerce. 
- **Built-in API routes.** Why even bother with Django/Python when you can stay inside the same  Next.js project and use [API Routes](https://nextjs.org/docs/api-routes/introduction)? Your code infra (tooling, linting, formatting) can now also stay inside the same project too. 
- **Built-in routing.** _You don't need to use React Router._ [File-system based routing](https://nextjs.org/docs/basic-features/pages) will achieve the same outcome, and it comes with Next.js by default.

**Reply by biplobmahadi** [reply]
I need my application fully single page application(no need to reload while routing). Is not it enough using **React, Redux, Axios, React-Router, antd** for frontend?
Is there any need of **nextjs**?

**Reply by kaizen-cmd** [reply]
if want your site to perform more faster, smoother and also good seo then next js is the way to go as it has auto code splitting (ultra fast loading without the refresh effect of the pages) and pages are rendered on the server side.
